### PR TITLE
feat: remove File.path

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change

It's time.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Removed nonstandard `File.path` extension from the Web `File` API.

### CODE REVIEW

**Review by erickzhao — APPROVED** [review]
docs: lgtm

Removing a `Custom DOM Elements` doc feels so good 🔥

**Comment by dsanders11** [comment]
Are we purposely not giving this a full deprecation cycle and going straight to yanking it?

**Comment by nornagon** [comment]
> Are we purposely not giving this a full deprecation cycle and going straight to yanking it?

It was quasi-deprecated in https://github.com/electron/electron/pull/38776, though it didn't get a breaking-changes.md callout.

**Comment by nornagon** [comment]
Moved the deletion to 32.x

**Review by ckerr — APPROVED** [review]
Thanks for this @nornagon and also thanks @MarshallOfSound for #38776

**Comment by ckerr** [comment]
API LGTM for 32.x deletion

**Review by ckerr — COMMENTED** [review]
API LGTM

**Review by jkleinsc — APPROVED** [review]
API LGTM

File: docs/breaking-changes.md [hunk 1]
```diff
@@ -12,6 +12,38 @@ This document uses the following convention to categorize breaking changes:
 * **Deprecated:** An API was marked as deprecated. The API will continue to function, but will emit a deprecation warning, and will be removed in a future release.
 * **Removed:** An API or feature was removed, and is no longer supported by Electron.
 
+## Planned Breaking API Changes (32.0)
+
+### Removed: `File.path`
+
+The nonstandard `path` property of the Web `File` object was added in an early version of Electron as a convenience method for working with native files when doing everything in the renderer was more common. However, it represents a deviation from the standard and poses a minor security risk as well, so beginning in Electron 31.0 it has been removed in favor of the [`webUtils.getPathForFile`](api/web-utils.md#webutilsgetpathforfilefile) method.
```
Comment (jkleinsc) [inline docs/breaking-changes.md:19]:
```suggestion
The nonstandard `path` property of the Web `File` object was added in an early version of Electron as a convenience method for working with native files when doing everything in the renderer was more common. However, it represents a deviation from the standard and poses a minor security risk as well, so beginning in Electron 32.0 it has been removed in favor of the [`webUtils.getPathForFile`](api/web-utils.md#webutilsgetpathforfilefile) method.
```

**Comment by MarshallOfSound** [comment]
@Makio64 Check the breaking changes doc: https://github.com/electron/electron/blob/main/docs/breaking-changes.md#removed-filepath

**Comment by xdmitrii** [comment]
@MarshallOfSound @erickzhao 

Is it possible to get file paths for all dropped files? I tried with example provided in the above [comment](https://github.com/electron/electron/pull/42053#issuecomment-2286980921):

```ts
for (const file of files) {
    const path = webUtils.getPathForFile(file);
    console.log(path);
}
```

But this method only works when I do:

```ts
const file = files[0];
```

Does this method work only for the first file in the list?

**Comment by Bug-Reaper** [comment]
@xdmitrii 
> @MarshallOfSound @erickzhao
> 
> Is it possible to get file paths for all dropped files? I tried with example provided in the above [comment](https://github.com/electron/electron/pull/42053#issuecomment-2286980921):
> 
> ```ts
> for (const file of files) {
>     const path = webUtils.getPathForFile(file);
>     console.log(path);
> }
> ```
> 
> But this method only works when I do:
> 
> ```ts
> const file = files[0];
> ```
> 
> Does this method work only for the first file in the list?

Sounds like some type of bug (or security measure) where it doesn't work when you reassign access via `const file`? I'd try a normal `for` loop:
```js
for (let i=0;i<files.length;i++){
  console.log(webUtils.getPathForFile(files[0]))
}
```
I actually need to handle multi-file eventually and can revisit this if I remember.

**Pull request merged**